hal/src/electron/modem/mdm_hal.cpp

```diff
@@ -1395,11 +1401,21 @@ bool MDMParser::checkNetStatus(NetStatus* status /*= NULL*/)
         if (RESP_OK != waitFinalResp(_cbCOPS, &_net, COPS_TIMEOUT)) {
             goto failure;
         }
-        // get the signal strength indication
-        sendFormated("AT+CSQ\r\n");
-        if (RESP_OK != waitFinalResp(_cbCSQ, &_net, CSQ_TIMEOUT)) {
+        // AT command used to collect signal stregnth is different for R410M radio
+        if (_dev.dev == DEV_SARA_R410) {
+            sendFormated("AT+UCGED?\r\n");
+            if (RESP_OK != waitFinalResp(_cbUCGED, &_net, UCGED_TIMEOUT)) {
```

Comment:
Currently, I am only checking this once and reading the RSRP/RSRQ values, and calling them final. 

There is a bit of information about this command. The command as per their AT manual, should give a response like this: 
```
AT+UCGED?
+RSRP: 473,1000,"-105.10",
+RSRQ: 473,1000,"-20.00",
OK
```

But in practice, based on if it's at the edge of towers or moving, is like the following, where it gives multiple responses on the RSRP/RSRQ string: 
```
AT+UCGED?
+RSRP: 473,1000,"-105.10",193,1000,"-100.80",
+RSRQ: 473,1000,"-20.00",193,1000,"-15.80",
OK
```
The goal is to be able to pick the value of the serving LTE cell. However, which cell (among 473 and 193 in the example above) is the serving cell is not obvious. We checked with u-blox, and they mentioned that the first set of values (473 in this example) is always _likely_ the serving cell. 
Currently, I am only collecting the RSRP/RSRQ values from the first cell id in the string. Another approach ( as Brett suggested ) is to run this command a few times every second, and make sure we are always on the same serving cell before finalizing the value? But I hear that, that could cause other processes to wait until this is complete?

Reply:
My suggestion here is if we see multiple values, to leave the RSRP and RSRQ as their last known value.  If we read multiple values for a second time, then perhaps we should set the values to one that indicates an error since we don't trust the readings any longer.  One problem with this approach is we don't currently control the retry interval.  A user could be calling Particle.publishVitals() every 1 second, or maybe every 30s.  So that persisted value could hang around for much longer than it should.

Another approach would be to force one retry in Device OS after a delay of 300ms if the first read contained multiple values.  This might not be long enough for the tower hand off to have occurred though, so some characterization may need to be done here with mobile testing.  One possible negative affect to this approach is occasionally the Particle.publishVitals() API will block for a bit longer while it tries to resolve this issue.

The reason why I like this approach is because in practice when mobile testing, when I would see multiple values reported for RSRP and RSRQ, I would still be able to publish data and it was received by the server.  So if we ultimately will be using RSRP and RSRQ to help qualify network operations (publishes, etc..), I would like to avoid being overly cautious and optimize for being overly optimistic (due to previous testing).  Reason being, if we can't send certain messages due to being overly cautious, it can cause an even longer delay while we tear down the network connection and rebuild it.

Without having an explicit interpretation of what multiple responses means, we must characterize it and implement a strategy for interpreting it that best suits our customers.

Reply:
Had a chat about this with Julien when you were in EU, and Julien also mentioned about Particle.publishvitals going to have to be blocked until the signal strength approach is resolved. I do not fully understand the potential solution here - I do not know if Particle.publishvitals is ok to be blocked for some time. 

On the other hand - FAE from ublox mentioned that the first value is likely from the serving LTE cell (which is the cell we should be collecting values from).
Further questions on this: 
- Is it ok to let Particle.publishvitals be blocked? 
- If yes, how many times should be retry at 300ms / how much time can that call be blocked?

Reply:
> On the other hand - FAE from ublox mentioned that the first value is likely from the serving LTE cell (which is the cell we should be collecting values from).
If we do some testing to corroborate that info, maybe that's the better method.  I could have sworn that I saw results that indicated the first value wasn't always the previous tower, or the next tower that the modem ended up selecting.

We should try our best not to block the call, so it might be something we create a 300ms+ timer for and process it on the system thread asynchronously.  I think one properly timed retry would be good.  Say we set it to 300ms, and keep getting multiple-"multiple responses" from UCGED.  Then the timing is not long enough for tower hand off.  300ms is the minimum retry time for UCGED per u-blox.  But perhaps it takes 500ms to reliably acquire the next tower... perhaps longer.  I know when I would poll every 2 seconds, I never saw multiple-multiples... so it's likely less than 2 seconds.

Reply:
So we ended up just trusting the first value in a multi-response.